client/layout/index.js

```diff
@@ -78,9 +78,16 @@ class _Layout extends Component {
 			isJetpackConnected,
 		} = this.props;
 
+		const navigationFlag = {
+			has_navigation: !! window.wcAdminFeatures.navigation,
```

Comment:
Definitely an edge case, but it's possible the feature is enabled, but the page we're on has opted out of the navigation experience.  `window.wcNavigation` should show us whether or not the navigation is on for the existing page.